src/js/player.js

```diff
@@ -314,6 +314,11 @@ vjs.Player.prototype.loadTech = function(techName, source){
 
   var techReady = function(){
     this.player_.triggerReady();
+
+    // add vjs-autoplay to hide poster and big-play-button if autoplay:true
+    if (this.options_['autoplay']) {
+      this.addClass('vjs-autoplay');
```

Comment:
Poster flashes even with HTML5 video element so I think the best way to handle the poster is not to have one if you are using autoplay.